contracts/governance/TrueFiVault.sol

```diff
@@ -61,6 +66,17 @@ contract TrueFiVault {
         _;
     }
 
+    function withdrawable(IERC20 token) public view returns (uint256) {
+        if (beneficiary == owner) {
+            return token.balanceOf(address(this));
+        }
+        uint256 timePassed = block.timestamp.sub(expiry.sub(DURATION));
+        if (timePassed > DURATION) {
+            timePassed = DURATION;
+        }
+        return token.balanceOf(address(this)).add(withdrawn[token]).mul(timePassed).div(DURATION).sub(withdrawn[token]);
```

Comment:
This system doesn't correctly enforce linear vesting: for example (ignoring staking rewards for a moment), assume someone started with 100 TRU in this vault. Then after half a year they could withdraw 50 TRU, tell the vault to stake the remaining 50, and then withdraw 25 stkTRU, effectively withdrawing 75% of the principal after only 50% of the time.

Reply:
Separately, what is the desired behavior as regards staking rewards? e.g. if someone starts with 100 TRU and after half a year has say 130 total thanks to staking rewards, is the desired behavior that they can withdraw 65: half their principal as well as half the rewards?

Reply:
Regarding withdrawing stkTRU - maybe we should just ban withdrawing stkTRU until the vault is opened? 
I think it's fine to allow withdrawing staking rewards